python-avd/pyavd/_eos_designs/shared_utils/filtered_tenants.py

```diff
@@ -302,19 +302,25 @@ def filtered_svis(
 
         Filtering based on accepted vlans since eos_designs_facts already
         filtered that on tags and trunk_groups.
+        Filter static_routes set under svis.
         """
         if not (self.network_services_l2 or self.network_services_l2_as_subint):
             return EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem.Svis()
 
         svis = vrf.svis._filtered(self.is_accepted_vlan)
 
-        # Handle svi_profile inheritance
-        svis = EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem.Svis([self.get_merged_svi_config(svi) for svi in svis])
-
         # Perform filtering on tags after merge of profiles, to support tags being set inside profiles.
         svis = svis._filtered(lambda svi: "all" in self.filter_tags or bool(set(svi.tags).intersection(self.filter_tags)))
+        all_svis = EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem.Svis()
+        for svi in svis:
```

Comment:
You should not change the order of things. Tag filtering needs to be done after the merge.

Reply:
Since you need to do more things now per SVI, you should probably just do your for-loop, merge the svi, look for tag filtering and `continue` if not passing the filter. Then do the static route extraction from the merged svi and finally append the merged svi to the output.

Reply:
It will probably also be quicker if you do the is_accepted_vlan check in there - before filtering.